snake_and_ladder.cpp

```diff
@@ -0,0 +1,118 @@
+#include<bits/stdc++.h>
+
+using namespace std;
+
+class Snake_and_ladder
+{
+	int s_no,l_no,p_no;
+	int v[101],s_head[101],s_tail[101],l_start[101],l_end[101];
+	string p_list[1001];
+	map<string,int> m;
+	public:
+    Snake_and_ladder()
+	{
+	  	s_no=0;
+	  	l_no=0;
+	  	p_no=0;
+	  	
+	  	for(int i=0;i<101;i++)
+	  	{
+	  		s_head[i]=0;
+	  		s_tail[i]=0;
+	  		l_start[i]=0;
+	  		l_end[i]=0;
+	  		v[i]=0;
+		}
+
+	}	
+    void s_insert(int s)
+	{
+		s_no=s;
+		
+		for(int i=1;i<=s_no;i++)
+		{
+			cin>>s_head[i]>>s_tail[i];
+			v[s_head[i]]=s_tail[i];
+		}
+
+	}	
+    void l_insert(int l)
+	{
+		l_no=l;
+		
+		for(int i=1;i<=l_no;i++)
+		{
+			cin>>l_start[i]>>l_end[i];
+			v[l_start[i]]=l_end[i];
+		}
+
+	}	
+	void p_insert(int p)
+	{
+		p_no=p;
+		
+		string str;
+		for(int i=1;i<=p;i++)
+	   {
+		cin>>p_list[i];
+		m[p_list[i]]=0;
+		
+	   }
+
+	}
+	void output()
+	{
+		int flag=0;
+	map<string,int>::iterator itr;
+	
+	itr=m.begin();
+
+	while(!flag)
+	{ 
+		int val=(rand()%6+1);
+		if(itr->second+val==100)
+		{
+			flag=1;
+			cout<<itr->first<<" wins the game";
+		}
+		else if(itr->second+val<100)
```

Comment:
what if there is a ladder that reaches 100?

Reply:
infinite loop?

Reply:
https://workattech.github.io/mock-machine-coding-1/problem-statement in this link it is mentioned that it is possible to win. Placing a snake at 100 will make it impossible to win.

Reply:
"what if there is a ladder that reaches 100"
No need to have a snake at 100.

Please dry run your code for that. There will be an infinite loop.